docs/src/main/paradox/routing-dsl/directives/security-directives/authenticateOAuth2.md

```diff
@@ -11,7 +11,7 @@
 
 ## Description
 
-Wraps the inner route with OAuth Bearer Token authentication support using a given @scala[`AuthenticatorPF[T]`]@java[`AuthenticatorPF<T>`].
+Wraps the inner route with OAuth Bearer Token authentication support using a given @scala[`Authenticator[T]`]@java[`Authenticator<T>`].
```

Comment:
For the Java API It looks like this directive takes a function from `Optional<ProvidedCredentials>` to `Optional<T>`.

For the Scala variation, would it be possible to use the `@scaladoc` directive to link to https://doc.akka.io/api/akka-http/current/akka/http/scaladsl/server/Directives$.html#Authenticator[T]=akka.http.scaladsl.server.directives.Credentials=%3EOption[T] ?

Reply:
sure, will do

Reply:
Lot more improvements done in https://github.com/akka/akka-http/pull/2518 as suggested, thanks!